services/storage/src/simcore_service_storage/simcore_s3_dsm.py

```diff
@@ -1249,7 +1252,11 @@ async def create_s3_export(
         *,
         progress_bar: ProgressBarData,
     ) -> StorageFileID:
-        source_object_keys: set[StorageFileID] = set()
+        source_object_keys: set[tuple[UserSelection, StorageFileID]] = set()
+
+        # ensure all selected items have the same parent
+        if not ensure_same_parent_in_user_selection(object_keys):
+            raise SelectionNotAllowedError(selection=object_keys)
 
```

Comment:
Question: why does storage need to care about this? If this is a frontend requirement, I am not sure this should be something so deep in the backend.. or maybe an option better. Cause from storage point of view, I don't really see the point.

Reply:
Since the request was to provide the files in the selection and not the full path path containing `/project_id/node_id/workspace`, when interested in exporting `workspace` this is a requirement, otherwise files from different services might collide and overwrite themselves. The simple and reliable solution is to raise an error.